metrics/prometheus.go

```diff
@@ -1716,6 +1717,16 @@ func (c *PrometheusCollector) collectContainersInfo(ch chan<- prometheus.Metric)
 				}
 			}
 		}
+		if c.includedMetrics.Has(container.PerfMetrics) {
+			for _, metric := range stats.PerfStats {
+				labelNames := append(labels, "cpu")
+				labelValues := append(values, strconv.Itoa(metric.Cpu))
```

Comment:
Would anyone ever want to aggregate over all events?  It may be better to give the metric a static name, e.g. perf_events_total, and add the event name as a label.  In general, putting non-constant things in metric names is an anti-pattern in prometheus.

Reply:
I use non-constant metric name only to pass scaling information. Here CPU ID is added as a label because I believe that passing raw values is better idea than aggregating them in a way that would suite all the use cases.

Reply:
For instance low value of scaling ration indicates that event was being measured for very short time and value might not be accurate. I had an internal discussion about aggregating perf stats this week and we were not able to find a solution that could be universal enough.

Reply:
I'm not quite sure what problems it could cause... I'm proposing

`perf_events_total{cpu=1, event="instruction"}`

That can be broken down in the same way that `container_perf_instructions{cpu=1}` can.  But I can also then write more generic queries, like `rate(perf_events_total[1m]) by (event)` to graph the rate of all perf events.  That way, someone could create a cAdvisor dashboard that is decoupled from their perf config, making it possible to re-use it.

What were the concerns you had about having a constant metric name?

Reply:
Got it. Looks perfectly reasonable to me now :)